src/main/scripts/src/app/shared/perspectives/collections/post-it/post-it-collections-perspective.component.ts

```diff
@@ -18,23 +18,87 @@
  * -----------------------------------------------------------------------/
  */
 
-import {Component, Input} from '@angular/core';
+import {Component, Input, OnInit} from '@angular/core';
+import {trigger, state, style, transition, animate, keyframes} from '@angular/animations';
+
 import {CollectionService} from '../../../../core/rest/collection.service';
 import {Perspective} from '../../perspective';
+import {Collection} from '../../../../core/dto/collection';
+import {LocalService} from '../../../../core/local.service';
 
 @Component({
   selector: 'post-it-collections-perspective',
-  templateUrl: './post-it-collections-perspective.component.html'
+  templateUrl: './post-it-collections-perspective.component.html',
+  styleUrls: ['./post-it-collections-perspective.component.scss'],
+  animations: [
+    trigger('animateVisible', [
+      state('in', style({height: '*', width: '*', opacity: 1})),
+      transition('void => *', [
+        animate(200, keyframes([
+          style({height: 0, width: 0, opacity: 0, offset: 0}),
+          style({height: '*', width: '*', opacity: 1, offset: 1})
+        ]))
+      ]),
+      transition('* => void', [
+        animate(200, keyframes([
+          style({height: '*', width: '*', opacity: 1, offset: 0}),
+          style({height: 0, width: 0, opacity: 0, offset: 1})
+        ]))
+      ])
+    ])
+  ]
 })
-export class PostItCollectionsPerspectiveComponent implements Perspective {
+export class PostItCollectionsPerspectiveComponent implements Perspective, OnInit {
 
   @Input()
   public query: string;
 
   @Input()
   public editable: boolean;
 
-  constructor(private collectionService: CollectionService) {
+  public placeholderTitle: string = 'Collection name';
+  public iconsPerPage: number = 36;
+  public newCollection: Collection = new Collection();
+  public collections: Collection[];
+  public pickerVisible: boolean = false;
+  public icons: string[];
+  public colors: string[];
+  public numbers: any[];
+
+  constructor(private collectionService: CollectionService,
+              private localService: LocalService) {
+    this.numbers = Array.apply(null, {length: this.iconsPerPage}).map(Number.call, Number);
   }
 
+  public ngOnInit(): void {
+    this.collectionService.getCollections()
+      .subscribe((collections: Collection[]) => {
+        this.collections = collections;
+      });
+    this.localService.getIcons().subscribe((icons: string[]) => {
+      this.icons = icons;
+    });
+    this.localService.getColors().subscribe((colors: string[]) => {
+      this.colors = colors;
+    });
+  }
+
+  public saveCollection() {
+    this.collectionService.saveCollection(this.newCollection)
+      .subscribe((code: String) => {
+        this.newCollection = new Collection();
+        this.collectionService.getCollections()
+          .subscribe((collections: Collection[]) => {
```

Comment:
Why do you subscribe again? It has already been done in the `ngOnInit()` method so it should react to changes.